bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,324 @@
+    print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+    print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+    # Write all keys back to a file
+    with open(all_keys_file, "w") as f_obj:
+        for video_key, item_list in this_all_keys_dict.items():
+            outstr = (
+                f"{video_key}{CSV_DELIMITER}{CSV_DELIMITER.join(map(str, item_list))}\n"
+            )
+            f_obj.write(outstr)
+
+    return this_all_keys_dict
+
+
+# From the keys present in NZSL, get all their ACL information
+def output_csv(this_all_keys_dict):
+    print(f"Getting ACLs for keys from S3 ({AWS_S3_BUCKET}) ...", file=sys.stderr)
+
+    # CSV header
+    csv_header_list = [
+        "Video S3 Key",
+        "Postgres ID",
+        "Gloss ID",
+        "Signbank Public",
+        "Expected S3 Canned ACL",
+        "Actual S3 Canned ACL",
+    ]
+    print(CSV_DELIMITER.join(csv_header_list))
+
+    for video_key, [
+        is_present,
+        db_id,
+        gloss_id,
+        is_public,
+    ] in this_all_keys_dict.items():
+
+        if not is_present:
+            print(f"{video_key},,,,,")
+            continue
+
+        # See signbank/video/models.py, line 59, in function set_public_acl()
+        canned_acl_expected = "public-read" if is_public else "private"
+        result = subprocess.run(
+            [
+                AWSCLIENT,
+                "s3api",
+                "get-object-acl",
+                "--output",
+                "json",
+                "--bucket",
+                AWS_S3_BUCKET,
+                "--key",
+                video_key,
+            ],
+            env=NEW_ENV,
+            shell=False,
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+        canned_acl = "unknown"
+        acls_grants_json = json.loads(result.stdout)["Grants"]
+        if len(acls_grants_json) > 1:
+            if (
+                acls_grants_json[0]["Permission"] == "FULL_CONTROL"
+                and acls_grants_json[1]["Permission"] == "READ"
+            ):
+                canned_acl = "public-read"
+        else:
+            if acls_grants_json[0]["Permission"] == "FULL_CONTROL":
+                canned_acl = "private"
+
+        # CSV columns
+        csv_column_list = [
+            f"{video_key}",
+            f"{db_id}",
+            f"{gloss_id}",
+            f"{is_public}",
+            f"{canned_acl_expected}",
+            f"{canned_acl}",
+        ]
+        print(CSV_DELIMITER.join(csv_column_list))
+
+
+print(f"Mode:        {args.mode}", file=sys.stderr)
+print(f"S3 bucket:   {AWS_S3_BUCKET}", file=sys.stderr)
+print(f"AWSCLIENT:   {AWSCLIENT}", file=sys.stderr)
+print(f"PGCLIENT:    {PGCLIENT}", file=sys.stderr)
+if "AWS_PROFILE" in NEW_ENV:
+    print(f"AWS profile: {NEW_ENV['AWS_PROFILE']}", file=sys.stderr)
+if "DATABASE_URL" in NEW_ENV:
+    print(f"DATABASE_URL:\n{NEW_ENV['DATABASE_URL']}", file=sys.stderr)
+
```

Comment:
issue: can we please stop outputting this? as it's guaranteed to have the database password in it, and makes me nervous that it'll mistakenly get captured in a screenshot or output copy 😅  

```suggestion
if "AWS_PROFILE" in NEW_ENV:
    print(f"AWS profile: {NEW_ENV['AWS_PROFILE']}", file=sys.stderr)

```

Reply:
Good point! Done.